### ISSUE TITLE
Private list not shown after delete & re-import

### ISSUE THREAD

**Issue by ilgrosso** [apache/incubator-ponymail#290]
Up until last week I used to regularly execute the following commands on one of our servers:

1. `mailx -f file.mbox` (to manually remove spam e-mails from archives)
1. `python3 edit-list.py --delete --source file.tirasa.net`
1. `python3 import-mbox.py --source  file.mbox --lid "file.tirasa.net" --html2text --private`

and everything has been always working fine.

This morning, after pulling the latest changes, the commands above still seem to work fine, but the private list `file@tirasa.net` is not neither shown nor reachable any more from the web UI.

* Anything to change in the commands above?
* How can I debug why the list is not reported nor accessible (e.g. when manually setting the URL to `/list.html?file@tirasa.net` a 302 to another list is returned)?

**Comment by sebbASF** [comment]
I just tried the same sequence and it worked OK for me.

I would try restarting the httpd server in case it is a caching issue.

Note that edit-list --delete does not remove the sources so the spam source will still be in the database. It won't be accessible externally.

**Comment by ilgrosso** [comment]
> I would try restarting the httpd server in case it is a caching issue.

Unfortunately, it's not cache.

> Note that edit-list --delete does not remove the sources so the spam source will still be in the database.

Really? Even if I remove the spam sources from the source mbox (step 1)?

Any debugging hint?

**Comment by sebbASF** [comment]
The --delete function only removes the mbox document. The mbox_source document will still be present in the database. AFAICT any attachments will also be kept.

Does the same problem exist with public mails?
e.g. try importing some test mails into a test list and repeating the process.

What happens if you import the de-spammed mbox into a brand-new list?

**Comment by ilgrosso** [comment]
> What happens if you import the de-spammed mbox into a brand-new list?

What do you mean? Simply `python3 import-mbox.py --source file.mbox --lid "file2.tirasa.net" --html2text --private` does not change anything: nor `file@tirasa.net` nor `file@tirasa.net` are neither visible nor accessible from PonyMail web UI.

**Comment by sebbASF** [comment]
That suggests that the problem is with file.mbox.
Can you import any mboxes successfully?

**Comment by ilgrosso** [comment]
> That suggests that the problem is with file.mbox.

Interesting: how could this be, with no error reported by `import-mbox.py`?

> Can you import any mboxes successfully?

Yes, I can: I have a cron job which does the same for all of our mailing lists, and I can see some e-mails imported today.

**Comment by sebbASF** [comment]
What _does_ import-mbox report?

What happens if you try to re-import the original mbox (with the spam in it)?

**Comment by ilgrosso** [comment]
> What does import-mbox report?

```
python3 import-mbox.py --source  file.mbox --lid "file.tirasa.net" --html2text --private
Checking that the database index ponymail exists ... 
Database exists OK
Doing file based import
Starting up to 1 threads to fetch the 1  lists
Thread-1: Thread started
Started no. 1
Thread-1: 1 elements left to slurp
Thread-1: Slurping file.mbox
Found attachment: Offerta-20150416-FC-00.pdf
<snip/>
Found attachment: error.Wf10.txt
Thread-1: Parsed 830 records (failed: 0) from file.mbox
Thread-1: Done, 0 elements left to slurp
All done! 830 records inserted/updated after 47 seconds. 0 records were bad and ignored
```

> What happens if you try to re-import the original mbox (with the spam in it)?

Where is that file located (I suppose it is the sources mentioned in the `delete` output as  `Action: Delete emails (sources will be kept!`)?

**Comment by ilgrosso** [comment]
I must add that when I query directly Elasticsearch via

`http://localhost:9200/ponymail/_search?_source_exclude=body&sort=date:desc`

I see in the returned JSON that `hits/hits` contains an e-mail from that `file` mbox in 8th place.

**Comment by ilgrosso** [comment]
Also - sorry for not having reported this before, but I've actually just noticed - that all private lists are not reported (when authenticated), hence the problem is not limited anyway to `file.mbox` and its mailing list.

**Comment by sebbASF** [comment]
What do you mean by "not reported"?

If there is an issue with auth, then I did update aaa.lua recently (it was badly broken).
Check that you have got the correct version of aaa.lua for your system.

**Comment by ilgrosso** [comment]
I confirm that restoring the former (customized) `site/api/lib/aaa.lua` makes everything work again. I'll stick with such version, until a fix for #292 is available.

Thanks for your support.
